### PR TITLE
fixed preview-description issue

### PR DESCRIPTION
**Pull request by leonPep42**

### CODE REVIEW

File: client-side/src/app/query-form/query-form.component.ts [hunk 1]
```diff
@@ -805,4 +807,8 @@ async previewDataHandler(data) {
 		await this.saveClicked();
 	}
 
+	isDataLimited(query: any): boolean {
+		return (query.Limitations?.License === "Free version" && query.Limitations?.TrialEndDate < new Date().toLocaleString())
```
Comment (saargt) [inline client-side/src/app/query-form/query-form.component.ts:811]: That's not how you compare date time in JS:
https://www.freecodecamp.org/news/javascript-date-comparison-how-to-compare-dates-in-js/

**Review by saargt — CHANGES_REQUESTED** [review]

File: client-side/src/app/query-form/query-form.component.ts (hunk 1 above)
Comment (leonPep42) [reply client-side/src/app/query-form/query-form.component.ts:811]: you mean I should use getTime?
we don't really need milliseconds comparison

File: client-side/src/app/query-form/query-form.component.ts [hunk 2]
```diff
@@ -17,7 +17,7 @@ import { VariableEditorComponent } from '../variable-editor/variable-editor.comp
 import { config } from '../addon.config';
 import { UserSelectComponent } from '../user-select/user-select.component';
 import { DataExportFormComponent } from '../data-export-form/data-export-form.component';
-
+import { licenseOptions } from 'src/metadata/license';
```
Comment (saargt) [inline client-side/src/app/query-form/query-form.component.ts:20]: Don't use absolute path, rather '../../metadata' or whatever.

File: client-side/src/app/query-form/query-form.component.ts [hunk 3]
```diff
@@ -808,7 +808,7 @@ async previewDataHandler(data) {
 	}
 
 	isDataLimited(query: any): boolean {
-		return (query.Limitations?.License === "Free version" && query.Limitations?.TrialEndDate < new Date().toLocaleString())
+		return (query.Limitations?.License === licenseOptions.Free && query.Limitations?.TrialEndDate?.getTime() < new Date().getTime())
```
Comment (saargt) [inline client-side/src/app/query-form/query-form.component.ts:811]: query.Limitations?.TrialEndDate will be a string, not a Date object you can .getTime() on.

**Review by saargt — CHANGES_REQUESTED** [review]

File: client-side/src/app/query-form/query-form.component.ts (hunk 2 above)
Comment (leonPep42) [reply client-side/src/app/query-form/query-form.component.ts:20]: first I did this and vs code yelled on me, fixed now :/

**Pull request merged**